sklearn/metrics/tests/test_classification.py

```diff
@@ -2252,6 +2254,94 @@ def test_brier_score_loss():
         brier_score_loss(['foo'], [0.4], pos_label='foo'), 0.36)
 
 
+@pytest.mark.parametrize('norm', ["l1", "l2", "max"])
+@pytest.mark.parametrize('strategy', ["uniform", "quantile"])
+def test_calibration_error_calibrated_predictions(norm, strategy):
+    ratio = 2
+    y_true = np.array([0, 0, 0, 1] + [0, 1, 1, 1])
+    y_pred = np.array([0.25, 0.25, 0.25, 0.25] + [0.75, 0.75, 0.75, 0.75])
+    assert_almost_equal(
+        calibration_error(y_true, y_pred, n_bins=ratio, norm=norm,
+                          strategy=strategy, reduce_bias=False),
+        0.)
+
+
+@pytest.mark.parametrize('norm', ["l1", "l2", "max"])
+@pytest.mark.parametrize('strategy', ["uniform", "quantile"])
+def test_calibration_error_uncalibrated_predictions(norm, strategy):
+    ratio = 2
+    y_true = np.array([0, 0, 0, 0] + [1, 1, 1, 1])
+    y_pred = np.array([0.25, 0.25, 0.25, 0.25] + [0.75, 0.75, 0.75, 0.75])
+    assert_almost_equal(
+        calibration_error(y_true, y_pred, n_bins=ratio, norm=norm,
+                          strategy=strategy, reduce_bias=False),
+        0.25)
+
+
+def test_calibration_error_sample_weights():
+    ratio = 2
+    y_true = np.array([0, 0, 0, 1] + [1, 1, 1, 1])
+    y_pred = np.array([0.25, 0.25, 0.25, 0.25] + [0.75, 0.75, 0.75, 0.75])
+    sample_weight = np.array([1, 1, 1, 1] + [3, 3, 3, 3])
+
+    assert_almost_equal(
```

Comment:
We are using the following pattern

```pytest
assert calibration_error(...) == pytest.approx(0.1875)
```
instead of `assert_almost_equal` since we introduced pytest